src/DataStructures/DataVector.hpp

```diff
@@ -54,15 +54,34 @@ using std::abs;  // NOLINT
 
 /*!
  * \ingroup DataStructuresGroup
- * \brief A class for storing data on a mesh.
+ * \brief Stores the nodal coefficients of a function on a Mesh.
  *
- * A Data holds an array of contiguous data and can be either owning (the array
- * is deleted when the Data goes out of scope) or non-owning, meaning it just
- * has a pointer to an array.
+ * \details Only use DataVector to represent function values on the
+ * computational grid. These nodal coefficients allow to approximate the
+ * function with interpolating Lagrange polynomials. Note that the DataVector
+ * only stores the function values, so interpreting the data requires knowledge
+ * of the corresponding grid points (nodes). These must be provided by an
+ * accompanying Mesh.
  *
- * A variety of mathematical operations are supported with DataVectors. In
- * addition to the addition, subtraction, multiplication, division, etc. there
- * are the following element-wise operations:
+ * A DataVector holds an array of contiguous data and can be either owning (the
+ * array is deleted when the Data goes out of scope) or non-owning, meaning it
+ * just has a pointer to an array.
+ *
+ * Unless you have a reason to use DataVector directly, you should generally use
+ * the Tensor type to represent Scalar functions or higher-rank tensors on the
+ * manifold. A Tensor makes use of one (Scalar) or more DataVectors to store
+ * its data.
+ *
+ * Refer to the documentation on [Data
+ * Structures](https://spectre-code.org/group__datastructuresgroup) for a list
```

Comment:
Don't hardcode our website; it prevents people from rendering their own documentation and having it work correctly.  I'm not sure what the syntax to do that is offhand.